### DISCUSSION TITLE
Make @js() directive work with blade component $attributes.

### DISCUSSION THREAD

**Discussion by tanthammar** [livewire/livewire#4083] [3. Feature ideas]
Is it possible to make the `@js()` directive work within blade component `$attributes`?
The @js directive is not rendered so Alpine never receives any data.

Example where it would be nice to make it work.

views/foo.blade.php
```blade
<x-button
   x-data="{ errors: @js($errors->all()) }"
   x-on:click="$dispatch('notify-errors', errors)"
   type="submit">
     Submit
</x-button>
```

components/button.blade.php
```blade
<button {{ $attributes->merge(['class' => 'classes']) }}>
{{ $slot }}
</button>
```

**Comment by joshhanley** [comment]
@tanthammar I think that is actually an issue with Laravel's blade compiler.

With the way the compiler works, using `{{ }}` inline doesn't work either. 

So this wouldn't work

```blade
<x-button test="{{ $test }}">...
```

I suspect that is also why the `@js` directive won't work.

I wonder whether there should a function called `js()` or something though, so you could use a php expression to generate it, like:

```blade
<x-button
   :x-data="'{ errors: ' . js($errors->all()) . ' }"
   x-on:click="$dispatch('notify-errors', errors)"
   type="submit">
     Submit
</x-button>
```

What do you think?

**Reply by tanthammar** [reply]
I made a PR #4108
